sdk/inc/azure/core/az_version.h

```diff
@@ -15,9 +15,8 @@
 #ifndef _az_VERSION_H
 #define _az_VERSION_H
 
-/// The version in string format used for telemetry following the `semver.org` standard
-/// (https://semver.org).
-#define AZ_SDK_VERSION_STRING "1.2.0-beta.1"
+#define _az_STRINGIFY2(x) #x
+#define _az_STRINGIFY(x) _az_STRINGIFY2(x)
```

Comment:
Why do we have two macros like this? Please add a comment.

Why isn't this sufficient?
```suggestion
#define _az_STRINGIFY(x) #x
```